ui/tests/e2e-framingham/test.js

```diff
@@ -0,0 +1,343 @@
+const JEST_TIMEOUT_MS = 120 * 1000;
+
+// Print test name at the beginning of each test
+jasmine.getEnv().addReporter({
+  specStarted: function(result) {
+    console.log(result.fullName);
+  }
+});
+
+describe("End-to-end framingham heart study teaching", () => {
+  beforeAll(async () => {
+    // It can take a while for servers to start up
+    jest.setTimeout(JEST_TIMEOUT_MS);
+    await waitForElasticsearchIndex();
+
+    // Hide snackbar because it prevents clicking on some facet bars
+    await page.goto("http://localhost:4400", { timeout: 120000 });
+    await page.waitForSelector("[class*='datasetName']");
+    await page.evaluate(() => {
+      localStorage.setItem("hasShownSnackbarv2", "true");
+    });
+  });
+
+  beforeEach(async () => {
+    await page.goto("http://localhost:4400", { timeout: 120000 });
+    await page.waitForSelector("[class*='datasetName']");
+  });
+
+  test("Age facet", async () => {
+    // Assert facet rendered correctly
+    await assertFacet("AGE", "80-89", ["0", "0", "9"]);
+
+    // Click on facet bar and assert page updated correctly
+    let facetBar = await getFacetBar("AGE", "60-69", "2");
+    await facetBar.click("");
+    await waitForFacetsUpdate(1083);
+    await assertFacet("AGE", "70-79", ["0", "231", "440"]);
+    await assertFacet("GLUCOSE", "400-449", ["0", "1", "0"]);
+
+    // Make sure unselected facet value bar is dimmed
+    facetBar = await getFacetBar("AGE", "50-59", "2");
+    const barColor = await page.evaluate(bar => bar.style.fill, facetBar);
+    // Vega translates our hex colors to rgb so it must be validated this way.
+    expect(barColor).toBe("rgb(191, 225, 240)");
+  });
+
+  test("Glucose facet", async () => {
```

Comment:
Fundamentally Glucose facet isn't different from Age facet right? Delete this test